loader/loader_test.go

```diff
@@ -238,6 +240,70 @@ var _ = Describe("Release Adapter", Ordered, func() {
 		})
 	})
 
+	When("calling GetPreviousRelease", func() {
+		var (
+			additionalRelease *v1alpha1.Release
+		)
+
+		BeforeEach(func() {
+			now := time.Now()
+
+			// Ensure the initial release is created
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{Name: release.Name, Namespace: release.Namespace}, release)
+			}).Should(Succeed())
+
+			// Delay to ensure a distinct timestamp
+			time.Sleep(2 * time.Second)
+
+			additionalRelease = &v1alpha1.Release{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:              "additional-release",
+					Namespace:         "default",
+					CreationTimestamp: metav1.Time{Time: now.Add(-1 * time.Hour)},
+				},
+				Spec: v1alpha1.ReleaseSpec{
+					ReleasePlan: releasePlan.Name,
+					Snapshot:    "snapshot-current",
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, additionalRelease)).To(Succeed())
+
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{Name: additionalRelease.Name, Namespace: additionalRelease.Namespace}, additionalRelease)
+			}).Should(Succeed())
+		})
+
+		AfterEach(func() {
+			Eventually(func() error {
+				return k8sClient.Delete(ctx, additionalRelease)
+			}).Should(Succeed())
+		})
+
+		It("returns the previous release if it exists", func() {
+			Eventually(func() error {
+				return k8sClient.Get(ctx, client.ObjectKey{Name: release.Name, Namespace: release.Namespace}, release)
+			}).Should(Succeed())
+
+			returnedObject, err := loader.GetPreviousRelease(ctx, k8sClient, additionalRelease)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(returnedObject).NotTo(BeNil())
+			Expect(returnedObject.Name).To(Equal(release.Name)) // Ensure it matches the initial release
+		})
+
+		It("returns a NotFound error if no previous release exists", func() {
```

Comment:
You have many conditions in the function
```
	for i, currentRelease := range releases.Items {
		if currentRelease.Name != release.Name &&
			currentRelease.CreationTimestamp.Time.Before(releaseTime) &&
			currentRelease.Spec.ReleasePlan == release.Spec.ReleasePlan { // Ensure the same ReleasePlan
			if previousRelease == nil || currentRelease.CreationTimestamp.Time.After(previousRelease.CreationTimestamp.Time) {
```
but you only test if a previous release exists or not. We need tests for two releases with different ReleasePlans, just one release in the namespace, and when there are more than 2 releases in the namespace with matching ReleasePlans